q2_demux/tests/test_demux.py

```diff
@@ -774,3 +774,48 @@ def test_phred_score_out_of_range(self):
             with open(plot_fp, 'r') as fh:
                 html = fh.read()
                 self.assertIn('<strong>Danger:</strong>', html)
+
+    def test_inconsistent_sequence_length_single(self):
+        sequences = [('@s1/1 abc/1', 'GGG', '+', 'YYY'),
+                     ('@s2/1 abc/1', 'CC', '+', 'PPP'),
+                     ('@s3/1 abc/1', 'AAA', '+', 'PPP'),
+                     ('@s4/1 abc/1', 'TTT', '+', 'PPP')]
+        bsi = BarcodeSequenceFastqIterator(self.barcodes, sequences)
+
+        barcode_map = pd.Series(['AAAA', 'AACC'], index=['sample1', 'sample2'])
+        barcode_map = qiime2.MetadataCategory(barcode_map)
+
+        demux_data = emp_single(bsi, barcode_map)
+        # TODO: Remove _PlotQualView wrapper
+        with tempfile.TemporaryDirectory() as output_dir:
+            with self.assertRaisesRegex(ValueError,
+                                        'Observed sequences of length'):
+                summarize(output_dir, _PlotQualView(demux_data,
+                                                    paired=False), n=2)
```

Comment:
I think this test is stochastic with `n=2`, since the `n` is associated with raw sequence reads, not necessarily number of samples. (e.g. I could randomly draw `s1/1 abc/1` and `s4/1 abc/1` and there wouldn't be an error.

Am I missing something here?